src/main/java/net/mcreator/ui/blockly/BlocklyJavascriptTemplates.java

```diff
@@ -27,7 +27,11 @@
 	public static String variableListExtension(VariableType variableType) {
 		return "Blockly.Extensions.register('" + variableType.getName() + "_variables',function () {"
 				+ "this.getInput(\"var\").appendField(new Blockly.FieldDropdown(getVariablesOfType(\""
-				+ variableType.getBlocklyVariableType() + "\")), 'VAR');" + "});";
+				+ variableType.getBlocklyVariableType() + "\")), 'VAR');"
+				+ "this.getField('VAR').setValidator(function (variable) {"

```

Comment:
What is the purpose of this validator? I did not work with Blockly validators before :D

Reply:
It's called when the selected value in the dropdown is changed, so that the block shape is updated

Reply:
Maybe make it so that when a player variable is selected, source/event entity is added by default?